### ISSUE TITLE
HTML comments: tests from `ckeditor5-paste-from-office` are failing

### ISSUE THREAD

**Issue by psmyrek** [ckeditor/ckeditor5#10155]
## Provide a description of the task

On HTML comment feature branch `ck/8822-html-comment` run `yarn test -f paste-from-office`.

There are multiple errors related to new view element `<$comment></$comment>`.

**Comment by psmyrek** [comment]
Maybe we could add a new filter that will remove all comments from pasted content?

**Comment by Reinmar** [comment]
IDK if this is going to solve the issue but I think that for now we can strip all comments on any kind of paste. I'd implement the "stripping" in the upcast converter based on the context of conversion  (clipboard uses `$clipboardHolder` as the conversion root).

**Comment by Reinmar** [comment]
The above is incorrect, or rather – insufficient.

The problem appears earlier – that the `<$comment>` nodes are present in the view after DOM->view conversion:

```html
      + expected - actual

      -<$comment></$comment>
      -<p dir="ltr" id="docs-internal-guid-f8b26bf1-7fff-40c0-af18-1234567890ab" style="line-height:1.38;margin-bottom:0pt;margin-top:0pt"><span style="background-color:transparent;color:#000000;font-family:Arial;font-size:11pt;font-style:normal;font-variant:normal;font-weight:400;text-decoration:none;vertical-align:baseline;white-space:pre-wrap">Hello world</span></p>
      -<$comment></$comment>
      
      +<p dir="ltr" id="docs-internal-guid-f8b26bf1-7fff-40c0-af18-1234567890ab" style="line-height:1.38;margin-bottom:0pt;margin-top:0pt"><span style="background-color:transparent;color:#000000;font-family:Arial;font-size:11pt;font-style:normal;font-variant:normal;font-weight:400;text-decoration:none;vertical-align:baseline;white-space:pre-wrap">Hello world</span></p>
```

So, even if we'd strip comments on paste via the converter, that will not fix PFO tests.

Therefore, we need a solution for these tests or this mechanism (meaning – PFO in general) that will:

*   fix the tests
*   ensure that PFO does not break as the existing code might not consider `<$comment>` elements in the view

My best idea so far is to remove comments as one of the normalization steps in PFO. So far we have two steps: MSW + GDocs. After that we'd have:

*   parse (extracted from MSW normalizer)
*   strip comments (new)
*   MSW
*   GDocs

**Comment by psmyrek** [comment]
In this issue I'm focused only on the `PasteFromOffice` plugin. Stripping all comments in any kind of paste will be handled in https://github.com/ckeditor/ckeditor5/issues/10127.

Currently, `PasteFromOffice` plugin allows only one normalizer to be used for data transformation, so I would prefer not to change this rule to fix the compatibility problem with comments. Instead of extracting the parser to a separate (mandatory) nomalizer, it would be simpler when data parsing occurs right after the first and only normalizer is found. Thus the main rule that only one normalizer can be used for data transformation will not be changed.

Additionally, in order not to duplicate the logic for removing comments in various places, it seems that the `DomConveter#domToView()` method should support a new (fourth) option to skip parsing the comments.

**Comment by ma2ciek** [comment]
Closed by #10206.
